desc/io/input_reader.py

```diff
@@ -776,6 +778,185 @@ def write_desc_input(filename, inputs, header=""):  # noqa: C901 - fxn too compl
+            path of the DESC output equilibrium file
+        header : str
+            text to print at the top of the file
+
+        """
+        # open the file, unless its already open
+        if not isinstance(filename, io.IOBase):
+            f = open(filename, "w+")
+        else:
+            f = filename
+        f.seek(0)
+
+        eq = load(inputs)
+        try:
+            eq0 = eq[-1]
+        except TypeError:
+            eq0 = eq
+
+        f.write(header + "\n")
+
+        f.write("# global parameters\n")
+        f.write("sym = {:1d} \n".format(eq0.sym))
+        f.write("NFP = {:3d} \n".format(int(eq0.NFP)))
+        f.write("Psi = {:.8f} \n".format(eq0.Psi))
+
+        f.write("\n# spectral resolution\n")
+        for key, val in {
+            "L_rad": "L",
+            "M_pol": "M",
+            "N_tor": "N",
+            "L_grid": "L_grid",
+            "M_grid": "M_grid",
+            "N_grid": "N_grid",
+        }.items():
+            f.write(key + " = {}\n".format(" " + str(eval("eq0.{0}".format(val)))))
+
+        print("DESC will automatically choose continuation parameters..\n")
+
+        f.write("\n\n# solver tolerances\n")
+        try:
+            f.write("ftol = {}\n".format(eq0._bdry_ratio))
+        except AttributeError:
+            print("ftol not given in the output. Setting to 1E-2...")
+            f.write("ftol = {}\n".format(float(1e-2)))
+
+        try:
+            f.write("xtol = {}\n".format(eq0._pres_ratio))
+        except AttributeError:
+            print("xtol not given in the output. Setting to 1E-6...")
+            f.write("xtol = {}\n".format(float(1e-6)))
+
+        try:
+            f.write("gtol = {}\n".format(eq0._curr_ratio))
+        except AttributeError:
+            print("gtol not given in the output. Setting to 1E-6...")
+            f.write("gtol = {}\n".format(float(1e-6)))
+
+        try:
+            f.write("maxiter = {}\n".format(eq0._pert_order))
+        except AttributeError:
+            print("maxiter not given in the output. Setting to 100...")
+            f.write("maxiter = {}\n".format(int(100)))
+
+        f.write("\n\n# solver methods\n")
+        try:
+            f.write("optimizer = {}\n".format(eq0._optimizer))
+        except AttributeError:
+            print("Optimizer absent in the output. Setting to lsq-exact...")
+            f.write("optimizer = {}\n".format("lsq-exact"))
+
+        try:
+            f.write("objective = {}\n".format(eq0._objective))
+        except AttributeError:
+            print("Objective not given in the output. Setting to force...")
+            f.write("objective = {}\n".format("force"))
+
+        f.write("spectral_indexing = {}\n".format(eq0._spectral_indexing))
+        f.write("node_pattern = {}\n".format(eq0._node_pattern))
+
+        f.write("\n# pressure and rotational transform/current profiles\n")
+
+        # can't import other stuff in io due to circular imports, so have to check by name
```

Comment:
You could also do a local import (inside the function, not at the top of the file)